server/src/test/java/io/crate/integrationtests/LogicalReplicationITest.java

```diff
@@ -46,6 +52,86 @@ private String defaultTableSettings() {
         return joiner.toString();
     }
 
+    @Test
+    public void test_create_publication_checks_owner_was_not_deleted_before_creation() {
+        executeOnPublisher("CREATE TABLE doc.t1 (id INT)");
+
+        String publicationOwner = "publication_owner";
+        executeOnPublisher("CREATE USER " + publicationOwner);
+        executeOnPublisher("GRANT AL TO " + publicationOwner);
+        executeOnPublisher("GRANT DQL, DML, DDL ON TABLE doc.t1 TO " + publicationOwner);
+        UserLookup userLookup = publisherCluster.getInstance(UserLookup.class);
+        User user = Objects.requireNonNull(userLookup.findUser(publicationOwner), "User " + publicationOwner + " must exist");
```

Comment:
I rewrote it as suggested and this case started to fail with NPE on `userLookup` and I recalled why I did it like this - 
 
next line after this is `executeOnPublisher("DROP USER " + publicationOwner);`
and then goes  
```
assertThrowsMatches(
            () ->  executeOnPublisherAsUser("CREATE PUBLICATION pub1 FOR TABLE doc.t1", user),
```

which requires User object to create a Session - and I keep reference to a User object before it gets dropped - otherwise it's not possible to `executeAs`